### PR TITLE
hikari pool support

### PR DESCRIPTION
**Pull request by shaharuk-yb**
Flag based hikari pool support in benchbase.

Use flag ```use_hikari_pool: true/false``` in input yaml or xml. Default value is ```false```

Optionally, you can set max pool size for hikari through property ```max_pool_size: <pool_size>```. Default max pool size = terminals

Example yaml:
```
type: YUGABYTE
driver: com.yugabyte.Driver
url: jdbc:yugabytedb://localhost:5433/yugabyte?sslmode=disable&ApplicationName=featurebench&reWriteBatchedInserts=true&load-balance=false
username: yugabyte
password:
use_hikari_pool: true
max_pool_size: 10

batchsize: 128
isolation: TRANSACTION_REPEATABLE_READ
loaderthreads: 1
terminals: 1
collect_pg_stat_statements: true
use_dist_in_explain : true

works:
    work:
        time_secs: 5
        active_terminals: 1
        rate: unlimited
        warmup: 0
microbenchmark:
    class: com.oltpbenchmark.benchmarks.featurebench.customworkload.YBDefaultMicroBenchmark
    properties:
        setAutoCommit: true
        create:
            - DROP TABLE IF EXISTS pkey_rangescan_for_INclause_tbl_1;
            - CREATE TABLE pkey_rangescan_for_INclause_tbl_1(col_bigint_id_1 bigint, col_bigint_1 bigint, col_bigint_2 bigint, col_bigint_3 bigint, col_bigint_4 bigint, col_bigint_5 bigint, col_float2_1 float(2), col_float2_2 float(2), col_float5_1 float(5), col_float5_2 float(5), col_boolean_1 boolean, col_varchar10_1 varchar(10), col_varchar100_1 varchar(100), col_varchar100_2 varchar(100), col_varchar500_1 varchar(500), PRIMARY KEY(col_bigint_id_1 ASC));
        cleanup:
            - DROP TABLE IF EXISTS pkey_rangescan_for_INclause_tbl_1;
        loadRules:
            - table: pkey_rangescan_for_INclause_tbl_
              count: 1
              rows: 1000
              columns:
                  - name: col_bigint_id_
                    count: 1
                    util: PrimaryIntGen
                    params: [1, 1000]
                  - name: col_bigint_
                    count: 5
                    util: RandomNumber
                    params: [1, 1000]
                  - name: col_float2_
                    count: 2
                    util: RandomNoWithDecimalPoints
                    params: [1, 1000, 2]
                  - name: col_float5_
                    count: 2
                    util: RandomNoWithDecimalPoints
                    params: [1, 1000, 5]
                  - name: col_boolean_
                    count: 1
                    util: RandomBoolean
                  - name: col_varchar10_
                    count: 1
                    util: RandomAString
                    params: [10, 10]
                  - name: col_varchar100_
                    count: 2
                    util: RandomAString
                    params: [100, 100]
                  - name: col_varchar500_
                    count: 1
                    util: RandomAString
                    params: [500, 500]
        executeRules:
            - workload: scanG4_pkey_INclause_rangescan_increasing_rows_1row
              run:
                  - name: pkey_INclause_rangescan_increasing_rows_1row_test
                    weight: 100
                    queries:
                        - query: select col_bigint_id_1 from pkey_rangescan_for_INclause_tbl_1 where col_bigint_id_1 IN (?)
                          bindings:
                              count: 1
                              util: RandomNumber
                              params: [ 1, 1000 ]
```

You can also use this with xml files by adding following in the xml config
```
<use_hikari_pool>true</use_hikari_pool>
<max_pool_size>10</max_pool_size>
```

### CODE REVIEW

File: src/main/java/com/oltpbenchmark/api/BenchmarkModule.java [hunk 1]
```diff
@@ -91,6 +104,20 @@ public final Connection makeConnection() throws SQLException {
...
 
+    private HikariDataSource hikariDataSource;
+
+    public void createDataSource() {
+        HikariConfig config = new HikariConfig();
+        config.setJdbcUrl(workConf.getUrl());
+        if (!StringUtils.isEmpty(workConf.getUsername())) {
+            config.setUsername(workConf.getUsername());
+            config.setPassword(workConf.getPassword());
+            config.setMaximumPoolSize(workConf.getTerminals());
```
Comment (hbhanawat) [inline src/main/java/com/oltpbenchmark/api/BenchmarkModule.java:115]: There are some extra properties in the tpcc code. Can they be set using the jdbc url?

Comment (hbhanawat) [inline src/main/java/com/oltpbenchmark/api/BenchmarkModule.java:115]: I think we should have a separate property for max pool size. It should be separate from the number of terminals.

File: src/main/java/com/oltpbenchmark/api/BenchmarkModule.java [hunk 2]
```diff
@@ -73,14 +75,25 @@ public abstract class BenchmarkModule {
...
     // --------------------------------------------------------------------------
     // DATABASE CONNECTION
     // --------------------------------------------------------------------------
 
     public final Connection makeConnection() throws SQLException {
-
+        if (workConf.getXmlConfig().getBoolean("use_hikari_pool", false)) {
+            LOG.info("Using Hikari-pool to create connection");
+            return hikariDataSource.getConnection();
+        }
```
Comment (hbhanawat) [inline src/main/java/com/oltpbenchmark/api/BenchmarkModule.java:96]: else if might be better for readability

File: src/main/java/com/oltpbenchmark/api/BenchmarkModule.java (hunk 1 above)
Comment (shaharuk-yb) [reply src/main/java/com/oltpbenchmark/api/BenchmarkModule.java]: TPCC does not take jdbc url as input in the workload config.
I have kept it same is of the default benchbase, rest of the properties can be set through jdbc url.

Comment (shaharuk-yb) [reply src/main/java/com/oltpbenchmark/api/BenchmarkModule.java]: added ```max_pool_size``` which will be taken from the input yaml/xml. If not provided, default will be set to terminals.

File: src/main/java/com/oltpbenchmark/api/BenchmarkModule.java (hunk 2 above)
Comment (shaharuk-yb) [reply src/main/java/com/oltpbenchmark/api/BenchmarkModule.java]: done

**Review by hbhanawat — APPROVED** [review]
Complete the testing before merging it

**Pull request closed without merging**